typescript/packages/plugins/ironclad/src/ironclad.service.ts

```diff
@@ -0,0 +1,735 @@
+                    });
+                }
+
+                // Repay
+                await walletClient.sendTransaction({
+                    to: LENDING_POOL_ADDRESS,
+                    abi: LENDING_POOL_ABI,
+                    functionName: "repay",
+                    args: [
+                        parameters.assetAddress,
+                        withdrawAmount,
+                        2,
+                        walletClient.getAddress(),
+                    ],
+                });
+
+                // After repayment, get updated debt from protocol
+                const updatedReserveData = await walletClient.read({
+                    address: PROTOCOL_DATA_PROVIDER_ADDRESS as `0x${string}`,
+                    abi: PROTOCOL_DATA_PROVIDER_ABI,
+                    functionName: "getUserReserveData",
+                    args: [parameters.assetAddress, walletClient.getAddress()],
+                });
+                remainingDebt = (updatedReserveData as { value: any[] })
+                    .value[2];
+                withdrawalCount++;
+            }
+
+            // After debt is cleared, withdraw any remaining deposited assets
+            const finalReserveData = await walletClient.read({
+                address: PROTOCOL_DATA_PROVIDER_ADDRESS as `0x${string}`,
+                abi: PROTOCOL_DATA_PROVIDER_ABI,
+                functionName: "getUserReserveData",
+                args: [parameters.assetAddress, walletClient.getAddress()],
+            });
+            const remainingDeposit = (finalReserveData as { value: any[] })
+                .value[0]; // aToken balance
+
+            if (remainingDeposit > 0n) {
+                // Withdraw all remaining deposits
+                await walletClient.sendTransaction({
+                    to: LENDING_POOL_ADDRESS,
+                    abi: LENDING_POOL_ABI,
+                    functionName: "withdraw",
+                    args: [
+                        parameters.assetAddress,
+                        remainingDeposit,
+                        walletClient.getAddress(),
+                    ],
+                });
+            }
+            return `Successfully unwound position in ${
+                withdrawalCount - 1
+            } loops`;
+        } catch (error) {
+            throw Error(`Failed to execute loop withdraw: ${error}`);
+        }
+    }
+
+    @Tool({
+        name: "monitor_loop_position_ironclad",
+        description: "Monitor health of a looped position on Ironclad",
+    })
+    async monitorLoopPosition(
+        walletClient: EVMWalletClient,
+        parameters: MonitorPositionParameters
+    ): Promise<{
+        totalCollateral: string;
+        totalBorrowed: string;
+        currentLTV: string;
+        healthFactor: string;
+        liquidationThreshold: string;
+    }> {
+        try {
+            const asset = parameters.tokenAddress;
+
+            const decimalsResult = await walletClient.read({
```

Comment:
No need to use the decimals here. You can return in base units and let the LLM convert those